src/app/component/user/components/profile/shopping-list/shopping-list.component.ts

```diff
@@ -22,11 +22,18 @@ export class ShoppingListComponent implements OnInit, OnDestroy {
   public getShoppingList(): void {
     this.profileSubscription = this.profileService.getShoppingList()
     .subscribe(
-      (shoppingListArr: ShoppingList[]) => this.shoppingList = shoppingListArr,
-      error => this.shoppingList = []
+      (shoppingListArr: ShoppingList[]) => this.shoppingList = shoppingListArr
     );
   }
 
+  public getShoppingListLength() {
+     if (this.shoppingList === undefined) {
+       return this.shoppingList = [];
```

Comment:
this line will not return length, which means it will error again

Reply:
Fixed